### ISSUE TITLE
Unable to run RunTSNE. Error in .check_tsne_params(nrow(X), dims = dims, perplexity = perplexity, : perplexity is too large for the number of samples

### ISSUE THREAD

**Issue by Weiwen1992** [satijalab/seurat#1345]
I have 2 datasets one with 3K cells the other with 8K cells. They all encountered this error.

**Comment by andrewhill157** [comment]
We also ran into this and it looks like it might be related to how the embeddings are being subsetted before being input into rtsne:
```
dim(object@reductions$pca[[, 1:10]])
```
gives the dimensions I would expect on an older version of Seurat 3 (29365b7d19c156f0cd293e9200f4d2fc7bf6db8c) -- `1000  10` where 1000 is number of cells. 

But in the version we recently downloaded the behavior is different, and I get `50  10`.

This holds true for the exact line used in Seurat as well: `RunTSNE(object = object[[, dims]] ...`

**Comment by andrewhill157** [comment]
Note the line I mentioned above is from `Seurat:::RunTSNE.DimReduc`

**Comment by andrewwbutler** [comment]
Hi, this should be resolved now. We were doing some restructuring on the dims of `DimReduc` objects that messed this up for a couple commits. Reinstalling the latest (ba0641d09972cb74f868943ce910477ec34072de) should fix the bug.

**Comment by andrewhill157** [comment]
Great, thanks!

**Comment by hepcat72** [comment]
I am running into this error.  I installed Seurat in R Studio for the first time yesterday.  It's version 3.1.5.  How can I tell if I have the build with this fix?

**Comment by andrewwbutler** [comment]
The fix above is included with the latest (3.1.5) release. Are able to share a reproducible example?

**Comment by hepcat72** [comment]
I think my issue was simply a small number of cells.  I was able to work around it by manually supplying a smaller value for perplexity, as suggested [here](https://github.com/cole-trapnell-lab/monocle-release/issues/45).  E.g.

```
seuset <- Seurat::RunTSNE(seuset, dims = 1:6, resolution = 0.5, perplexity = 2)
```

And yes, 2 was the largest value that would not produce an error.  My small test dataset had only 7 cells.
